projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts

```diff
@@ -97,16 +97,35 @@ export default function reducer( state = INITIAL_STATE, action ) {
 		}
 
 		case ACTION_INCREASE_AI_ASSISTANT_REQUESTS_COUNT: {
-			// Increase request count;
 			const requestsCount = state.features.aiAssistant.requestsCount + action.count;
 
+			const currentTierValue = state.features.aiAssistant.currentTier?.value;
+
+			const isFreeTierPlan =
+				( typeof currentTierValue === 'undefined' && ! state.features.aiAssistant.hasFeature ) ||
+				currentTierValue === 1;
+
+			const isUnlimitedTierPlan =
+				( typeof currentTierValue === 'undefined' && state.features.aiAssistant.hasFeature ) ||
+				currentTierValue === 1;
+
+			const requestsLimit =
```

Comment:
I think here we should use:

- `state.features.aiAssistant.requestsLimit` for the free plan
- `Infinity` for the unlimited plan
- `state.features.aiAssistant.currentTier?.limit` for a plan with tier info